sklearn/decomposition/dict_learning.py

```diff
@@ -778,14 +773,14 @@ def dict_learning_online(X, n_components=2, alpha=1, n_iter=100,
             dt = (time.time() - t0)
             print('done (total time: % 3is, % 4.1fmn)' % (dt, dt / 60))
         if return_n_iter:
-            return code, dictionary.T, ii - iter_offset + 1
+            return code, dictionary.T, errors, ii - iter_offset + 1
         else:
-            return code, dictionary.T
+            return code, dictionary.T, errors
 
     if return_n_iter:
-        return dictionary.T, ii - iter_offset + 1
+        return dictionary.T, errors, ii - iter_offset + 1
     else:
-        return dictionary.T
+        return dictionary.T, errors
```

Comment:
unfortunately this is public API that we cannot break, we need to return the exact same signature as before. We can add a default argument (return_errors=False), but this logic is going to get messy